### ISSUE TITLE
Inline Document Settings

### ISSUE THREAD

**Issue by ricardo-dematos** [check-spelling/check-spelling#39]
Does [check-spelling/check-spelling](https://github.com/check-spelling/check-spelling/) supports [Inline Document Settings](https://cspell.org/configuration/document-settings/) ?

I'm trying with 
```python
func_name = "interp1d"  # cspell: disable-line
```
but it still reports has an unrecognized word.

**Comment by jsoref** [comment]
I've added this entry which should do what you want:
https://github.com/check-spelling/spell-check-this/blob/fd1b1e0c05bfd7b9fff7d9cd062a13a39d13b683/.github/actions/spelling/candidate.patterns#L6-L8

Note that it'd also do it for `# cspell: disable-line-breaks` which is perhaps not correct (maybe it needs a `\s*$`?)

You can copy that entry into `candidate.patterns` and then check-spelling should suggest the same item for `patterns.txt`, or you could copy it straight into `patterns.txt`.

**Comment by jsoref** [comment]
Draft changes for each:
* [exclude git blame ignore revisions file](https://github.com/check-spelling/spell-check-this/commit/f0c0f2194d62bb32651709fafc0824ade86df451)
* [exclude python lint configuration file](https://github.com/check-spelling/spell-check-this/commit/18c8e9f3dcd692bcefe19a40d8c074e91a551ad4)
* [exclude coverage configuration file](https://github.com/check-spelling/spell-check-this/commit/ae8454d37284742e914b496694c3b34018a9eb68)
* [exclude python project configurations](https://github.com/check-spelling/spell-check-this/commit/1c05f568e10a052b1af92a9bc21d264aa3973521) 
* [exclude python requirements files](https://github.com/check-spelling/spell-check-this/commit/6137f32cd78793ca349e807c6c8801f327516793)
* [Add additional way to ignore spelling for a line](https://github.com/check-spelling/spell-check-this/commit/e40301fe3866a307f45cb913f830c8c4b5b76eb8)

Notes:
* I don't include comments in the excludes file because the tool sorts the file automatically.
* I can't write `TOML` in the commit message because the repository is configured to check spelling and an allow list would involve putting `TOML` into everyone's defaults, which feels like a bit much.

**Comment by jsoref** [comment]
Note: For both check-spelling and spell-check-this, I'll rewrite any commits along `prerelease` until they're included in `main`.

So, I might, e.g. capitalize `Exclude`. Or harmonize the other bits...

**Comment by jsoref** [comment]
Hmm, they're case-insensitive -- I was worried about that. Doing that right w/ what I have is a bit complicated (or at the very least a bit ugly, as `cspell` would become `[Cc][Ss][Pp][Ee][Ll]{2}`.

There is a way to define case insensitive things in perl, but about half the time it has resulted in the regex engine getting really mad at me when it runs into fancy unicode content. v0.0.21 has a mode where it would more or less reject files that have unicode content, but some folks *like* using unicode, so they wouldn't use that flag.

---

For some of the other things, it's a lot harder...

At the moment, check-spelling is entirely line oriented. It retains roughly 0 state between lines, and its pattern format is similarly designed with that assumption. Just thinking about a way to enabling users to define their own state machines gives me a headache.

OTOH, defining a _simple_ state machine for pausing scanning and resuming is something I imagine I'll implement relatively soon (w/in a few releases). That'd be enough to support `/* spell-checker: enable */` / `/* cSpell:disable */`.

Consider:

```
teh qwick /* cSpell:disable */ brown brown /* spell-checker: enable */ foxx /* spell-checker: disable */
jumpss over over /* spell-checker: enable */ thhe /* cSpell: enable */ leighzy
dog
```

* Do I split each section first by enable/disable chunks? Offhand, I think the answer is yes
* Do I treat each enabled section as if it were at the start of the line? Or do I treat them as if they have my standard replacement character masking their context? -- Offhand, I think I'd go w/ the latter.

For reference, check-spelling currently does ~1-5 passes per line (patterns, forbidden patterns, check-spelling,, candidates, check-candidates-for-progress). Adding an extra pass (probably before patterns) wouldn't be the end of the world, but...

---

I've been thinking about [plugging check-spelling into vscode](https://github.com/check-spelling/check-spelling/wiki/Feature%3A-Visual-Studio-Code-plugin) for a while, and yeah, I'm aware that it has a cspell compatible thing atm.

**Comment by jsoref** [comment]
Yes. Note that it's just a pattern that you add to your local `patterns.txt` (and then use in your code).

**Comment by Piedone** [comment]
I see, thank you. What about excluding the next line due to the reasons I mention?

**Comment by jsoref** [comment]
That would require me to make the system stateful (#9). It's a lot more complicated -- amongst the many problems is how do I define a syntax which is backwards compatible -- the current `patterns.txt` rule format wouldn't really support that. I'm tempted to use a new file name and then I'd have to define a syntax. (In addition to actually writing the code to implement the stateful side for parsing files based on that data.)

I appreciate it's desired. And I think I'll try to implement it for the coming release.

**Comment by Piedone** [comment]
Thanks for your quick replies!
